### ISSUE TITLE
Buttons in confirmation modals cannot be reached with a screen reader

### ISSUE THREAD

**Issue by Arkalex** [Sage/carbon#7034]
### Description

Buttons in Confirmation Dialogs cannot be reached by a screen reader user unless they enter down twice into the modal. This leaves screen reader users at a significant disadvantage, as they may not enter down again into the modal, and have no way of confirming the action that is present in the modal.

### Reproduction

https://carbon.sage.com/?path=/docs/confirm--docs

### Steps to reproduce

1. Activate screen reader (cmd + F5 for MacOS)
2. Try to navigate to the actions (ctrl + option + left/right arrows)

### JIRA ticket numbers (Sage only)

SBS-112821

### Carbon version

144.0.0

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
